### ISSUE TITLE
Changing any addon options breaks UI

### ISSUE THREAD

**Issue by taro8** [CBATeam/CBA_A3#1027]
**Arma 3 Version:** `1.84` (stable)
**CBA Version:** `3.9.0` (stable)

**Mods:**
```
- CBA_A3
```

**Description:**
If you change any addon setting during a mission the UI breaks. Some options have wrong colors, map often displays the "No satellite connection" on top of the map. Restarting the game fixes the issue, but it means I cannot change options during a mission as UI breaks (map breaking is the most annoying part).

I thought it might be a broken profile so I nuked everything related to Arma 3 from the documents and started from fresh, same issues.

https://imgur.com/a/YETfP0Y

**Steps to reproduce:**
Start a mission, go into addon options, change something, go back to the game, open the map or quit to the main menu.

**Where did the issue occur?**
- Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**RPT log file:**
https://pastebin.com/XGU6REwW

**Comment by Cyruz143** [comment]
I've also noted this happening. Never made the connection to CBA settings changes though!

**Comment by jonpas** [comment]
Same, never thought of that!

**Comment by commy2** [comment]
Okay, here I write down my thinking and how I debugged this. May or may not be interesting.

So I could reproduce this issue by going straight into the editor, previewing a mission with only one playable unit placed, opening the ADDON OPTIONS in the Pause Menu/Esc Menu, and confirming directly without changing anything by clicking "OK". After a huge lag, when I now opened the map, the header bar was white. When I now returned to the main menu, "Single Player" was highlighted in blue and the popup menu entries were partly transparent.
Nothing happened when I did the same setup when I pressed "CANCEL" instead of "OK".
One needs to restart the game to "unbug" the main menu, but the mission can be restored by going back into 3den.

So what I tried is to short circuit the OK button by turning it into another Cancel button by commenting out the "confirm script". That is in fnc_initDisplayGameOptions.sqf in cba_settings, line 178f.
I restarted the game and tried the same thing. The bug was still there and it still had this huge lag. Of course it does, because the game still applies the "changes" from the "Configure Base" side of the menu, the vanilla GAME OPTIONS menu!

So now I started the game without CBA and instead opened the GAME OPTIONS menu and pressed OK.
Exactly the same issue as described happened.
Therefore I conclude that this is an effect from the GAME OPTIONS menu from BI and only happens when you use CBA's ADDON OPTIONS, because those are an extension of the GAME OPTIONS. This bug is entirely "BI's fault".

Now to circumvent this happening as often, I could simply disable the GAME OPTIONS' OK button whenever we open the ADDON OPTIONS instead of the GAME OPTIONS. That means that the bug would only happen, when you open the ADDON OPTIONS indirectly via GAME OPTIONS -> CONFIGURE ADDONS. But then we can just say that the bug would've happened anyway if you were to change any "game options".
